BJS2-84363-sms_notification

Linempy opened the pull request:
SMS нотификации

Linempy committed:
BJS2-84363-sms_notification
- написал реализацию smsService
- реализовал получение ивента RecommendationRequestedEvent
- написал Unit-тесты

RinatBeybutov commented on src/main/java/faang/school/notificationservice/config/redis/RedisConfig.java:
```diff
@@ -0,0 +1,69 @@
+
+    @Bean
+    public RedisMessageListenerContainer redisContainer(MessageListenerAdapter listener) {
+        log.info("ДОЛЖНЫ ВЫЗЫВАТЬСЯ");
```
странный комментарий, лучше что-то вроде :
Инициализация Redis контейнера

RinatBeybutov replied:
не исправлено

RinatBeybutov commented on src/main/java/faang/school/notificationservice/listener/AbstractMessageListener.java:
```diff
@@ -0,0 +1,63 @@
+                );
+    }
+
+    @Retryable(maxAttempts = 3, backoff = @Backoff(delay = 1000, multiplier = 2))
```
классно, что использовал ретрай

Linempy committed:
BJS2-84363-sms_notification
- поправил замечания
- изменил конфиг редис
  - сделал десериализацию через jackson2Json

Linempy committed:
BJS2-84363-sms_notification
- checkstyle
